driver/sccb.c

```diff
@@ -204,3 +204,57 @@ uint8_t SCCB_Write16(uint8_t slv_addr, uint16_t reg, uint8_t data)
     }
     return ret == ESP_OK ? 0 : -1;
 }
+
+uint16_t SCCB_Read_Addr16_Val16(uint8_t slv_addr, uint16_t reg)
+{
+    uint16_t data = 0;
+    uint8_t *data_u8 = (uint8_t *)&data;
+    esp_err_t ret = ESP_FAIL;
+    uint16_t reg_htons = LITTLETOBIG(reg);
+    uint8_t *reg_u8 = (uint8_t *)&reg_htons;
+    i2c_cmd_handle_t cmd = i2c_cmd_link_create();
+    i2c_master_start(cmd);
+    i2c_master_write_byte(cmd, ( slv_addr << 1 ) | WRITE_BIT, ACK_CHECK_EN);
+    i2c_master_write_byte(cmd, reg_u8[0], ACK_CHECK_EN);
+    i2c_master_write_byte(cmd, reg_u8[1], ACK_CHECK_EN);
+    i2c_master_stop(cmd);
+    ret = i2c_master_cmd_begin(sccb_i2c_port, cmd, 1000 / portTICK_RATE_MS);
+    i2c_cmd_link_delete(cmd);
+    if(ret != ESP_OK) return -1;
+
+    cmd = i2c_cmd_link_create();
+    i2c_master_start(cmd);
+    i2c_master_write_byte(cmd, ( slv_addr << 1 ) | READ_BIT, ACK_CHECK_EN);
+    i2c_master_read_byte(cmd, &data_u8[1], ACK_VAL);
+    i2c_master_read_byte(cmd, &data_u8[0], NACK_VAL);
+    i2c_master_stop(cmd);
+    ret = i2c_master_cmd_begin(sccb_i2c_port, cmd, 1000 / portTICK_RATE_MS);
+    i2c_cmd_link_delete(cmd);
+    if(ret != ESP_OK) {
+        ESP_LOGE(TAG, "W [%04x]=%04x fail\n", reg, data);
+    }
+    return data;
+}
+
+uint8_t SCCB_Write_Addr16_Val16(uint8_t slv_addr, uint16_t reg, uint16_t data)
+{
+    esp_err_t ret = ESP_FAIL;
+    uint16_t reg_htons = LITTLETOBIG(reg);
+    uint8_t *reg_u8 = (uint8_t *)&reg_htons;
+    uint16_t data_htons = LITTLETOBIG(data);
+    uint8_t *data_u8 = (uint8_t *)&data_htons;
+    i2c_cmd_handle_t cmd = i2c_cmd_link_create();
+    i2c_master_start(cmd);
+    i2c_master_write_byte(cmd, ( slv_addr << 1 ) | WRITE_BIT, ACK_CHECK_EN);
+    i2c_master_write_byte(cmd, reg_u8[0], ACK_CHECK_EN);
+    i2c_master_write_byte(cmd, reg_u8[1], ACK_CHECK_EN);
+    i2c_master_write_byte(cmd, data_u8[0], ACK_CHECK_EN);
+    i2c_master_write_byte(cmd, data_u8[1], ACK_CHECK_EN);
+    i2c_master_stop(cmd);
+    ret = i2c_master_cmd_begin(sccb_i2c_port, cmd, 1000 / portTICK_RATE_MS);
+    i2c_cmd_link_delete(cmd);
+    if(ret != ESP_OK) {
+        ESP_LOGE(TAG, "W [%04x]=%04x fail\n", reg, data);
+    }
+    return ret == ESP_OK ? 0 : -1;
```

Comment:
Maybe switch return type to `esp_err_t`, because currently you have it as `uint8_t`, but here you are returning `-1`

Reply:
I guess this would need to be done for the other `write` methods as well

Reply:
The current sensor driver code has many expressions similar to `if(SCCB_Write() < 0)`, so although the return value type can be defined as `esp_err_t` in the function declaration, in the relevant code, we only check whether the return value of the function is less than 0. Therefore, there are two options.   
1) We only define the return value type of the function as `esp_err_t` in the function declaration, and do not really return the return value of the `i2c_master_cmd_begin()`.
2) Or we can directly define the return value of this function as type `int`.

Reply:
in that case, using `int` or `int8_t` would also do the job

Reply:
The return value of `sccb_write()` has been changed to type `int`.

Reply:
@me-no-dev ,At present, it is not easy to buy some cameras that support JPEG encoding. A new sensor driver can work normally, but the current submission is required. If there is no problem, please help merge the submission. We will submit the driver code of the new camera as soon as possible. thanks.